gateway/ln-gateway/tests/integration_tests.rs

```diff
@@ -145,13 +145,19 @@ pub fn sha256(data: &[u8]) -> sha256::Hash {
     bitcoin::hashes::sha256::Hash::hash(data)
 }
 
+/// Test helper function for paying a valid BOLT11 invoice with a gateway
+/// specified by `gateway_id`.
 async fn pay_valid_invoice(
     invoice: Bolt11Invoice,
     user_client: &ClientArc,
-    client: &ClientArc,
+    gateway_client: &ClientArc,
+    gateway_id: &PublicKey,
 ) -> anyhow::Result<()> {
-    // User client pays test invoice
+    // Set the active gateway to be the gateway with id `gateway_id`
     let user_lightning_module = &user_client.get_first_module::<LightningClientModule>();
+    user_lightning_module.set_active_gateway(gateway_id).await?;
```

Comment:
Sorry, should have been more clear: a few lines down you call `pay_invoice`:

https://github.com/fedimint/fedimint/blob/f26de71c1ed23565cfcb34934183961eed58898e/gateway/ln-gateway/tests/integration_tests.rs#L59-L65

Internally that already just fetches a gateway (I don't even know if it still cares about the active gateway concept, i.e. if this PR would improve anything). If we don't fetch a random gateway but can supply one then we don't have to call `set_active_gateway` here anymore and get one step closer to removing it altogether.

Reply:
I add the function `select_gateway` which should be helpful moving forward in removing the active gateway concept. This PR fixes the flaky test, I'm going to work on removing the active gateway concept next.